### ISSUE TITLE
More frequent Rust releases

### ISSUE THREAD

**Issue by iajoiner** [delta-io/delta-rs#969]
# Description

I just found that the current main branch is quite different from the last Rust release (0.4.0). Maybe it is a good idea to do more frequent releases? I can help with the release process since I'm already helping with releases at arrow-rs.

**Use Case**

**Related Issue(s)**

**Comment by wjones127** [comment]
Yes, [we've planned to release more frequently](https://github.com/delta-io/delta-rs/blob/main/CONTRIBUTING.md#roadmap-2022h2) but haven't quite yet. @houqp is preparing a new release now.

**Comment by houqp** [comment]
We just released 0.5.0 to crates.io! Agreed that we should do these more frequently.

**Comment by iajoiner** [comment]
@wjones127 @houqp Thanks! Please let me know whether you guys need help doing biweekly releases.

**Comment by houqp** [comment]
Thanks @iajoiner ! We could use some help for organizing the rust releases now. How about we start to organize new rust releases in dedicated github issues going forward? In the issue, we can discuss:

* Version
* Whether there are MRs that are blocking the release
* Better update summary in changelog (optional)

**Comment by iajoiner** [comment]
@houqp You're welcome! I can start filing them every two weeks starting from Dec 9th. Changelogs can be automatically generated.

**Comment by iajoiner** [comment]
https://github.com/delta-io/delta-rs/issues/971
Here we go! Please let me know if the schedule is ok..
